orchestration/dap_orchestration/solids.py

```diff
@@ -0,0 +1,122 @@
+        "output_prefix": String,
+        "end_time": String,
+        "target_class": String,
+        "api_token": String,
+    }
+)
+def extract_records(context: AbstractComputeExecutionContext) -> str:
+    """
+    :return: Returns the path to extracted files.
+    """
+    arg_dict = {
+        "pullDataDictionaries": "true" if context.solid_config["pull_data_dictionaries"] else "false",
+        "outputPrefix": context.solid_config["output_prefix"],
+        "endTime": context.solid_config["end_time"],
+        "target_class": context.solid_config["target_class"],
+        "apiToken": context.solid_config["api_token"],
+        "scala_project": "dog-aging-hles-extraction"
+    }
+    context.resources.beam_runner.run(arg_dict)
+    # todo: resolve mypy error: Returning Any from function declared to return "str"
+    return context.solid_config["output_prefix"]
+
+# todo: create a builder method that returns a SolidDefintion
+def _build_extract_config(config: dict[str, str], target_class: str) -> dict[str, str]:
+    return {
+        "pull_data_dictionaries": config["pull_data_dictionaries"],
+        "output_prefix": config["output_prefix"],
+        "end_time": config["end_time"],
+        "api_token": config["api_token"],
+        "target_class": target_class
+    }
+
+
+@configured(extract_records)
+def hles_extract_records(config: dict[str, str]) -> dict[str, str]:
+    return _build_extract_config(config, "org.broadinstitute.monster.dap.hles.HLESurveyExtractionPipeline")
+
+
+@configured(extract_records)
+def cslb_extract_records(config: dict[str, str]) -> dict[str, str]:
+    return _build_extract_config(config,"org.broadinstitute.monster.dap.cslb.CslbExtractionPipeline")
+
+
+@configured(extract_records)
+def env_extract_records(config: dict[str, str]) -> dict[str, str]:
+    return _build_extract_config(config, "org.broadinstitute.monster.dap.environment.EnvironmentExtractionPipeline")
+
+
+@solid(
+    required_resource_keys={"beam_runner"},
+    config_schema={
+        "output_prefix": String,
+        "target_class": String,
+    }
+)
+def transform_records(context: AbstractComputeExecutionContext, input_prefix: str) -> str:
+    """
+    :return: Returns the path to the transformation output json files.
+    """
+    arg_dict = {
+        "inputPrefix": input_prefix,
+        "outputPrefix": context.solid_config["output_prefix"],
+        "target_class": context.solid_config["target_class"],
+        "scala_project": "dog-aging-hles-transformation"
+    }
+    context.resources.beam_runner.run(arg_dict)
+    # todo: resolve mypy error: Returning Any from function declared to return "str"
+    return context.solid_config["output_prefix"]
+
+def _build_transform_config(config: dict[str, str], target_class: str) -> dict[str, str]:
+    return {
+        "output_prefix": config["output_prefix"],
+        "target_class": target_class
+    }
+
+
+# todo: how do I specify the input_values (output_prefix) for a configured function?
```

Comment:
🤔 Is this still an issue? I believe it should be passed in via the config at runtime, but I'm not 100% on that. Happy to pair up and figure this out.